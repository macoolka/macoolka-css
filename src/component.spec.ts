
import { Rule,OutputProps,parseRuleJSS,parseRule,theme } from '.';

/**
 * convert number type to px or precent
 * @getter
 */
export type EProp = {
    size: 'small' | 'medium' | 'large' | 'xlarge'
};
export type SProp = {
    disabled: boolean
};

export type Props = OutputProps<EProp & SProp>;

export const rule: Rule<SProp, EProp> = {
    style: {
        mkLayout: 'inlineRow',
        borderColor: 'currentColor',
        width:20,
        mkTransition:  'background-color, color, opacity',
        selector: {
            '& > svg': {
                fill: 'currentColor',
            },
        },
    },
    rule: {
        disabled: (a: boolean) => ({
            opacity: a ? 0.38 : 1,
        }),
    },
    ruleEnum: {
        size: {
            small: ()=>({
                mkSquare: 'small',
                borderWidth: 1,
                selector: {
                    '& > svg': {
                        mkSquare: 'small',
                    },
                },
            }),
            medium:()=>({
                mkSquare: 'medium',
                borderWidth: 1,
                mkMedia:[{mkWidth:'small'},{mkWidth:'medium'},{mkWidth:'large'},{mkWidth:'xlarge'}],
                selector: {
                    '& > svg': {
                        mkSquare: 'medium',
                    },
                },
            }),
            large: ()=>({
                borderWidth: 2,
                mkSquare: 'large',
                selector: {
                    '& > svg': {
                        mkSquare: 'large',
                    },
                },
            }),
            xlarge: ()=>({
                borderWidth: 2,
                mkSquare: 'xlarge',
                selector: {
                    '& > svg': {
                        mkSquare: 'xlarge',
                    },
                },
            }),
        },
    },
};

const parse = parseRuleJSS(rule)(theme);
const parseString = parseRule(rule)(theme);
describe('icon', () => {
    it('parse icon', () => {
        expect(parse({
            size: 'small',
        })).toMatchSnapshot();
        expect(parse({
            mkMedia:[{mkWidth:'small'},{mkWidth:'medium'},{mkWidth:'large'},{mkWidth:'xlarge'}],
            size: 'small',
        })).toMatchSnapshot();
         expect(parse({
            size: 'medium',
            disabled:true,
        })).toMatchSnapshot(); 
        expect(parseString({
            size: 'small',
        })).toMatchSnapshot();
        expect(parseString({
            mkMedia:[{mkWidth:'small'},{mkWidth:'medium'},{mkWidth:'large'},{mkWidth:'xlarge'}],
            size: 'small',
        })).toMatchSnapshot();
         expect(parseString({
            size: 'medium',
            disabled:true,
        })).toMatchSnapshot();        
    })

});