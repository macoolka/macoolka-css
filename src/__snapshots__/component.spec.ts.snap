// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse component rule 1 should parse empty props 1`] = `
"font-size: 48px;
:focus {
  width: 30px;
}"
`;

exports[`should parse component rule 2 should parse common props 1`] = `
"font-size: 48px;
height: 180px;
:focus {
  width: 30px;
}"
`;

exports[`should parse component rule 3 should parse common props for place rule props 1`] = `
"font-size: 30px;
:focus {
  width: 30px;
}"
`;

exports[`should parse component rule 4 should parse component props for palce default style  1`] = `
"font-size: 96px;
:focus {
  width: 10px;
}
&:hover {
  color: #f50057;
}"
`;

exports[`should parse component rule 5 should parse common props and component props using the order base prop>module prop > mix prop>component prop >component default style 1`] = `
"font-size: 7px;
letter-spacing: -.01562em;
line-height: 6rem;
font-weight: 300;
font-family: Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif;
text-decoration: inherit;
text-transform: inherit;
:focus {
  width: 10px;
}
&:hover {
  color: #f50057;
}"
`;

exports[`should parse component rule 5 should parse common props and component props using the order base prop>module prop > mix prop>component prop >component default style 2`] = `
"font-size: 14px;
letter-spacing: -.01562em;
line-height: 6rem;
font-weight: 300;
font-family: Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif;
text-decoration: inherit;
text-transform: inherit;
:focus {
  width: 10px;
}
&:hover {
  color: #f50057;
}"
`;

exports[`should parse component rule 5 should parse common props and component props using the order base prop>module prop > mix prop>component prop >component default style 3`] = `
"font-size: 10px;
:focus {
  width: 10px;
}
&:hover {
  color: #f50057;
}"
`;

exports[`should parse component rule 5 should parse common props and component props using the order base prop>module prop > mix prop>component prop >component default style 4`] = `
"font-size: 96px;
:focus {
  width: 10px;
}
&:hover {
  color: #f50057;
}"
`;

exports[`should parse component rule 5 should parse common props and component props using the order base prop>module prop > mix prop>component prop >component default style 5`] = `
"font-size: 48px;
:focus {
  width: 30px;
}"
`;

exports[`should parse component rule 6 should parse media props 1`] = `
"font-size: 48px;
padding: 4px;
@media screen and (max-width: 80em) {
  padding: 8px;
}
:focus {
  width: 30px;
}"
`;

exports[`should parse component rule 7 should parse media props in rule 1`] = `
"font-size: 60px;
padding: 4px;
@media screen and (max-width: 80em) {
  padding: 8px;
}
:focus {
  width: 10px;
}"
`;
